contracts/pricefeeds/MultiplicativePriceFeed.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.15;
+
+import "../vendor/@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "../IPriceFeed.sol";
+
+/**
+ * @title Multiplicative price feed
+ * @notice A custom price feed that multiplies the prices from two price feeds and returns the result
+ * @author Compound
+ */
+contract MultiplicativePriceFeed is IPriceFeed {
+    /** Custom errors **/
+    error BadDecimals();
+    error InvalidInt256();
+
+    /// @notice Version of the price feed
+    uint public constant override version = 1;
+
+    /// @notice Description of the price feed
+    string public override description;
+
+    /// @notice Number of decimals for returned prices
+    uint8 public immutable override decimals;
+
+    /// @notice Chainlink price feed A
+    address public immutable priceFeedA;
+
+    /// @notice Chainlink price feed B
+    address public immutable priceFeedB;
+
+    /// @notice Combined scale of the two underlying Chainlink price feeds
+    int public immutable combinedScale;
+
+    /// @notice Scale of this price feed
+    int public immutable priceFeedScale;
+
+    /**
+     * @notice Construct a new multiplicative price feed
+     * @param priceFeedA_ The address of the first price feed to fetch prices from
+     * @param priceFeedB_ The address of the second price feed to fetch prices from
+     * @param decimals_ The number of decimals for the returned prices
+     * @param description_ The description of the price feed
+     **/
+    constructor(address priceFeedA_, address priceFeedB_, uint8 decimals_, string memory description_) {
+        priceFeedA = priceFeedA_;
+        priceFeedB = priceFeedB_;
+        uint8 priceFeedADecimals = AggregatorV3Interface(priceFeedA_).decimals();
+        uint8 priceFeedBDecimals = AggregatorV3Interface(priceFeedB_).decimals();
+        combinedScale = signed256(10 ** (priceFeedADecimals + priceFeedBDecimals));
+
+        if (decimals_ > 18) revert BadDecimals();
+        decimals = decimals_;
+        description = description_;
+        priceFeedScale = int256(10 ** decimals);
+    }
+
+    /**
+     * @notice Price for the latest round
+     * @return roundId Round id from price feed B
+     * @return answer Latest price
+     * @return startedAt Timestamp when the round was started; passed on from price feed B
```

Comment:
The only weirdness to this approach is that you could call this function twice, get the same "round id" and timestamps, but a different price (if the feed for price A had updated since your last call, but the feed for price B had not)

Reply:
Yeah I think it's impossible to avoid any weirdness here. The only thing Comet cares about is the price, so we could eventually just use `latestAnswer()` instead of `latestRoundData()`.

If we eventually want to improve the oracle and check timestamps for staleness, that work would be done in the price feed so Comet really never needs to know about timestamp, round id, etc.